skglm/estimators.py

```diff
@@ -133,16 +133,26 @@ def _glm_fit(X, y, model, datafit, penalty):
                 "The size of the WeightedL1 penalty weights should be n_features, "
                 "expected %i, got %i." % (X_.shape[1], len(penalty.weights)))
 
+    # TODO Handle multi-task case
     if is_classif:
-        solver = cd_solver  # TODO to be be replaced by an instance of BaseSolver
+        if isinstance(datafit, Logistic):
```

Comment:
if you do this you still need to pass all arguments to model.

Instead, add a `solver` argument to `_glm_fit`.

Users will instantiate their solver, their penalty and their datafit that they will pass to `GeneralizedLinearEstimator`.

Reply:
The goal is to avoid potentially long if elif like this if we have 10 solvers.